chain/indexer/src/streamer/streamer.rs

```diff
@@ -215,16 +230,30 @@ pub(crate) async fn start(
         };
 
         let latest_block_height = block.header.height;
-        if last_synced_block_height == 0 {
-            last_synced_block_height = latest_block_height;
-        }
+        let start_syncing_block_height = if let Some(last_synced_block_height) =
+            last_synced_block_height
+        {
+            last_synced_block_height + 1
+        } else {
+            match indexer_config.sync_mode {
+                crate::SyncModeEnum::FromInterruption => {
+                    match db.get(b"last_synced_block_height").unwrap() {
+                        Some(value) => String::from_utf8(value).unwrap().parse::<u64>().unwrap(),
+                        None => latest_block_height,
+                    }
+                }
+                crate::SyncModeEnum::LatestSynced => latest_block_height,
+                crate::SyncModeEnum::BlockHeight(height) => height,
+            }
+        };
+
         debug!(
             target: INDEXER,
             "The last synced block is #{} and the latest block is #{}",
-            last_synced_block_height,
+            last_synced_block_height.unwrap(),
```

Comment:
This will crash since `last_synced_block_height` is not updated up until the end of the loop. Just report that the streaming is about to start for block height `start_syncing_block_height` and going towards `latest_block_height`.